statsmodels/regression/tests/test_bayes_mixed_glm.py

```diff
@@ -0,0 +1,377 @@
+
+            rslt1 = glmm1.fit_map()
+
+            for k in range(3):
+
+                if k == 0:
+                    vb_mean = rslt1.params
+                    vb_sd = np.ones_like(vb_mean)
+                elif k == 1:
+                    vb_mean = np.zeros(len(vb_mean))
+                    vb_sd = np.ones_like(vb_mean)
+                else:
+                    vb_mean = np.random.normal(size=len(vb_mean))
+                    vb_sd = np.random.uniform(1, 2, size=len(vb_mean))
+
+                mean_grad, sd_grad = glmm1.vb_elbo_grad(vb_mean, vb_sd)
+
+                def elbo(vec):
+                    n = len(vec) // 2
+                    return glmm1.vb_elbo(vec[:n], vec[n:])
+
+                x = np.concatenate((vb_mean, vb_sd))
+                g1 = approx_fprime(x, elbo, 1e-5)
+                n = len(x) // 2
+
+                mean_grad_n = g1[:n]
+                sd_grad_n = g1[n:]
+
+                assert_allclose(mean_grad, mean_grad_n, atol=1e-2,
+                                rtol=1e-2)
+                assert_allclose(sd_grad, sd_grad_n, atol=1e-2,
+                                rtol=1e-2)
+
+
+def test_simple_logit_vb():
+
+    y, exog_fe, exog_vc, ident = gen_simple_logit(10, 10, 0)
+    exog_vc = sparse.csr_matrix(exog_vc)
+
+    glmm1 = BinomialBayesMixedGLM(y, exog_fe, exog_vc, ident, vcp_p=0.5,
+                                  fe_p=0.5)
+    rslt1 = glmm1.fit_map()
+
+    glmm2 = BinomialBayesMixedGLM(y, exog_fe, exog_vc, ident, vcp_p=0.5,
+                                  fe_p=0.5)
+    rslt2 = glmm2.fit_vb(rslt1.params)
+
+    rslt1.summary()
+    rslt2.summary()
+
+    assert_allclose(rslt1.params[0:5], np.r_[
+        0.75330405, -0.71643228, -2.49091288, -0.00959806,  0.00450254],
+                    rtol=1e-4, atol=1e-4)
+
+    assert_allclose(rslt2.params[0:5], np.r_[
+        0.79338836, -0.7599833, -0.64149356, -0.24772884,  0.10775366],
+                    rtol=1e-4, atol=1e-4)
+
+
+def test_simple_poisson_vb():
+
+    y, exog_fe, exog_vc, ident = gen_simple_poisson(10, 10, 1)
+    exog_vc = sparse.csr_matrix(exog_vc)
+
+    glmm1 = PoissonBayesMixedGLM(y, exog_fe, exog_vc, ident, vcp_p=0.5)
+    rslt1 = glmm1.fit_map()
+
+    glmm2 = PoissonBayesMixedGLM(y, exog_fe, exog_vc, ident, vcp_p=0.5)
+    rslt2 = glmm2.fit_vb(rslt1.params)
+
+    rslt1.summary()
+    rslt2.summary()
+
+    assert_allclose(rslt1.params[0:5], np.r_[
+        -0.07233493, -0.06706505, -0.47159649,  1.12575122, -1.02442201],
+                    rtol=1e-4, atol=1e-4)
+
+    assert_allclose(rslt1.cov_params.flat[0:5], np.r_[
+        0.00790914, 0.00080666, -0.00050719, 0.00022648, 0.00046235],
+                    rtol=1e-4, atol=1e-4)
+
+    assert_allclose(rslt2.params[0:5], np.r_[
+        -0.07088814, -0.06373107, -0.22770786,  1.12923746, -1.26161339],
+                    rtol=1e-4, atol=1e-4)
+
+    assert_allclose(rslt2.cov_params[0:5], np.r_[
+        0.00747782, 0.0092554, 0.04508904, 0.02934488, 0.20312746],
```

Comment:
where do the reference numbers come from?
regression test or from other package?

Reply:
It's regression testing.  I'm not aware of any other implementation of this method for GLIMMIX models.